src/beast/beast/nudb/detail/field.h

```diff
@@ -165,18 +174,32 @@ read (istream& is, U& u)
     u = t;
 }
 
+// read field from istream
+
+template <class T, class U>
+void
+read (istream& is, U& u)
+{
+    readp<T>(is.data(field<T>::size), u);
+}
+
 // write field to ostream
 
 template <class T, class U, std::enable_if_t<
+    std::is_same<T, std::uint8_t>::value>* = nullptr>
+void
+write (ostream& os, U const& u)
+{
+    std::uint8_t* p =
+        os.data(field<T>::size);
+    *p = u;
```

Comment:
My first thought was "I didn't know ostream had a member called data". It's a little confusing to see classes with the same name as classes from the std namespace. I'll have to adjust my thinking when looking at files in beast.